### PR TITLE
Feat no external section conns

### PR DESCRIPTION
**Pull request by joshuef**
Ahead of #1994

### CODE REVIEW

**Comment by joshuef** [comment]
bors try

**Comment by joshuef** [comment]
bors try

File: sn_comms/src/error.rs [hunk 1]
```diff
@@ -16,6 +16,9 @@ pub type Result<T, E = Error> = std::result::Result<T, E>;
 #[derive(Debug, Error)]
 #[allow(missing_docs)]
 pub enum Error {
+    /// Any unknown node comms should be bidi, initiated by the other side
+    #[error("Attempted to create a connection to an unkown node")]
+    CreatingConnectionToUnknownNode,
```
Comment (bochaco) [inline sn_comms/src/error.rs:21]: I thin it'll help when debugging to have the peer's info as part of the error

Comment (joshuef) [reply sn_comms/src/error.rs:21]: oh yeh, dejavu from my old PR. :D I'll update that

**Review by bochaco — APPROVED** [review]

**Pull request merged**